datalake/src/main/java/com/sequenceiq/datalake/service/sdx/SdxService.java

```diff
@@ -1353,10 +1355,14 @@ public Set<String> getInstanceGroupNamesBySdxDetails(SdxClusterShape clusterShap
         return result;
     }
 
-    public FlowIdentifier rotateSaltPassword(SdxCluster sdxCluster) {
+    public FlowIdentifier rotateSaltPassword(SdxCluster sdxCluster, RotateSaltPasswordReason reason) {
         FlowIdentifier flowIdentifier = ThreadBasedUserCrnProvider.doAsInternalActor(
                 regionAwareInternalCrnGeneratorFactory.iam().getInternalCrnForServiceAsString(),
-                initiatorUserCrn -> stackV4Endpoint.rotateSaltPasswordInternal(WORKSPACE_ID_DEFAULT, sdxCluster.getCrn(), initiatorUserCrn)
+                initiatorUserCrn -> {
+                    RotateSaltPasswordRequest request = new RotateSaltPasswordRequest();
+                    request.setReason(reason);
```

Comment:
nit: can you make a static method for this request like `RotateSaltPasswordRequest.withReason` or similar to avoid multiple line lambda expression?

Reply:
can you check this one too?

Reply:
I have added it as a constructor parameter